[ljy0422] iP

ljy0422 opened the pull request:
### **Jiayou**

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Jiayou frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/ljy0422/ip/releases/tag/A-Jar)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😉

Features:

- [x]  Managing tasks
- [x]  Managing deadlines (coming soon)
- [x]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
level-0

Commit:
level 1 echo

Commit:
level 2 add list

Commit:
level 3 mark as done

Commit:
level 4 todo, event, ddl

Commit:
A-textuitesing

Commit:
level 5 handle errors

Commit:
level 6 delete

Commit:
A-Enums

Commit:
level 7

Commit:
level 8

Commit:
more-oop

Commit:
more-opp

Commit:
package

ljy0422 committed:
Merge branch 'add-gradle-support'

ljy0422 committed:
level 9

ljy0422 committed:
Add JavaDoc to all the classes and non-trivial methods

ljy0422 committed:
Create ToDoTest class to test ToDo class
There are three tests that individually test different methods in the ToDo class.

ljy0422 committed:
Create EventTest class to test Event class
There are three tests that individually test different methods in the Event class.

ljy0422 committed:
Create DeadlineTest class to test Deadline class
There are three tests that individually test different methods in the Deadline class.

ljy0422 committed:
Package the app as an executable JAR file

ljy0422 committed:
Fix the typo in the file name of the output JAR file

ljy0422 committed:
Fix the bug in the Storage class
In the original load method of the Storage class all the tasks are loaded into the task list with undone status.

alfaloo commented on src/main/java/jiayou/Jiayou.java:
```diff
@@ -0,0 +1,41 @@
+package jiayou;
+
+import jiayou.task.*;
```
Do you think wildcard imports should be avoided?

alfaloo commented on src/main/java/jiayou/Parser.java:
```diff
@@ -0,0 +1,92 @@
+package jiayou;
+
+import java.time.LocalDate;
+import jiayou.task.*;
```
Again, can you try importing without wildcard? (its discouraged in the course coding standard)

alfaloo reviewed (commented):
Great code! very well done and good attention to details with regards to coding standards. The only things I noticed were the wildcard imports which are discouraged in this course. Keep up with the good work!

ljy0422 committed:
Add a guide function to the Ui class
The guide function is to introduce all the command types and usages of the chatbot to the user.

ljy0422 committed:
Fix some minor style problems
I improved some parts of the code according to the coding standard.

ljy0422 committed:
Edit some method names according to the naming standard

ljy0422 committed:
Fix some minor coding style problems according to the feedback of CheckStyle

ljy0422 committed:
Add a preliminary GUI to the chatbot

ljy0422 committed:
Add preliminary command parsing to the GUI of chatbot

ljy0422 committed:
Improve the appearance of GUI

h4ow3i commented on src/main/java/jiayou/Parser.java:
```diff
@@ -0,0 +1,107 @@
+     * @param tasks the task list of the chatbot.
+     * @param input the input string to be parsed.
+     */
+    public void parse(TaskList tasks, String input) {
```
could consider naming the method "parseCommand" to be more clear.

h4ow3i reviewed (commented):
Hi! I am the first reviewer, however it seems the second reviewer has already commented on your coding standards. Thus I'll be commenting on your code quality instead. Overall, good and clear codes, had little to no trouble understanding.

ljy0422 committed:
Utilize varargs in some command processing methods

h4ow3i commented on src/main/java/jiayou/task/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+ * @author Liu Jiayao
+ */
+public class Deadline extends Task {
+    private LocalDate byTime;
```
Could consider using "due" or "by" to be more clear as the LocalDate variable could be a date with no time

ljy0422 committed:
Edit the names of some methods and variables

ljy0422 committed:
Utilize varargs in some command processing methods

ljy0422 committed:
Utilize varargs in some command processing methods

ljy0422 committed:
Add greeting and farewell messages to the GUI

ljy0422 committed:
Improve the appearance of GUI

ljy0422 committed:
Improve the appearance of GUI

ljy0422 committed:
Package files in a more organized way

ljy0422 committed:
Use assert feature in some methods

This commit introduces the use of assertions in the markTask, unmarkTask, addTask, deleteTask methods to enhance the reliability and maintainability of our task management functionality. By asserting the expected state of tasks after some operations, we can catch logical errors early in the development process.

In `markTask`, we now assert that the task's status is true after it has been marked as done. This ensures that the task's state accurately reflects the action performed by the method.

Similarly, in `unmarkTask`, we assert that the task's status is false after it has been marked as undone. This validation acts as a safeguard against potential discrepancies between the task's intended and actual states.

We add an assertion in `addTask` to verify that the task list contains the newly added task. This ensures the successful addition of tasks and aids in the early detection of issues related to task storage.

We also add an assertion in `deleteTask` to verify that the task list doesn't contain the deleted task any more. This ensures the successful deletion of tasks.

ljy0422 committed:
Merge pull request #1 from ljy0422/branch-A-Assertions

Use assert feature in some methods

ljy0422 committed:
Improve code quality in my code

I use blank lines to separate groups of related statements, to structure the code logically.

I try to comply with SLAP, extracting codes from existing methods to another new method, to make the levels of abstraction in the methods clearer.

ljy0422 committed:
Merge pull request #2 from ljy0422/branch-A-CodeQuality

Improve code quality in my code

ljy0422 committed:
Use streams in two methods

ljy0422 committed:
Merge pull request #3 from ljy0422/branch-A-Streams

Use streams in two methods

ljy0422 committed:
Add rescheduling functionality to the chatbot

ljy0422 committed:
Merge pull request #4 from ljy0422/branch-B-Snooze

Add rescheduling functionality to the chatbot

ljy0422 committed:
Set up CI for the project

ljy0422 committed:
Set up CI for the project

ljy0422 committed:
Delete duplicate files

ljy0422 committed:
Delete duplicate files

ljy0422 committed:
Set up CI for the project

ljy0422 committed:
Display error message in a red color

ljy0422 committed:
Improve the code quality

ljy0422 committed:
Improve the code quality

ljy0422 committed:
Improve the code quality

ljy0422 committed:
Improve the code quality

ljy0422 committed:
Add more auto unit tests

ljy0422 committed:
Fix some coding style issues

ljy0422 committed:
Update README.md

ljy0422 committed:
Update README.md

ljy0422 committed:
Update README.md

ljy0422 committed:
Update README.md

ljy0422 committed:
Add the Ui image to docs

ljy0422 committed:
Merge branch 'master' of https://github.com/ljy0422/ip

ljy0422 committed:
Update README.md

ljy0422 committed:
Update README.md

ljy0422 committed:
Update README.md

ljy0422 committed:
Update README.md

ljy0422 committed:
Update README.md

ljy0422 committed:
Update README.md
